src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/CertificateGenerator.cs

```diff
@@ -286,14 +293,39 @@ private X509CertificateContainer CreateCertificate(bool isAuthority, X509Certifi
 
             if (!isAuthority)
             {
-                var subjectAlternativeNames = new Asn1Encodable[subjects.Length];
-
-                for (int i = 0; i < subjects.Length; i++)
+                if (isMachineCert)
                 {
-                    subjectAlternativeNames[i] = new GeneralName(GeneralName.DnsName, subjects[i]);
+                    var subjectAlternativeNames = new Asn1Encodable[subjects.Length];
+                    
+                    // All endpoints should also be in the Subject Alt Names 
+                    for (int i = 0; i < subjects.Length; i++)
+                    {
+                        // Machine certs can have additional DNS names
+                        subjectAlternativeNames[i] = new GeneralName(GeneralName.DnsName, subjects[i]);
+                    }
+
+                    _certGenerator.AddExtension(X509Extensions.SubjectAlternativeName, true, new DerSequence(subjectAlternativeNames));
+                }
+                else
+                {
+                    if (subjects.Length > 1)
+                    {
+                        var subjectAlternativeNames = new Asn1EncodableVector();
+                    
+                        // Only add a SAN for the user if there are any
+                        for (int i = 1; i < subjects.Length; i++)
+                        {
+                            Asn1EncodableVector otherNames = new Asn1EncodableVector();
+                            otherNames.Add(new DerObjectIdentifier("1.3.6.1.4.1.311.20.2.3"));
```

Comment:
Nit: This mystical string of numbers could use a comment.  Or a 'const' with a descriptive name.